cinq_auditor_required_tags/providers.py

```diff
@@ -125,32 +125,38 @@ def terminate_ec2_instance(client, resource):
             error
         ))
 
+
 def delete_s3_bucket(client, resource):
     try:
         session = get_aws_session(resource.account)
         bucket = session.resource('s3', resource.location).Bucket(resource.resource_id)
 
-        lifecyclePolicy = {'Rules': [
-            {'Status': 'Enabled',
-                'NoncurrentVersionExpiration': {u'NoncurrentDays': 1},
-                'Filter': {u'Prefix': ''},
-                'Expiration': {u'Days': 1},
-                'AbortIncompleteMultipartUpload': {u'DaysAfterInitiation': 1},
-                'ID': 'cloudInquisitor'}]}
+        lifecycle_policy = {
+            'Rules': [
+                {'Status': 'Enabled',
+                 'NoncurrentVersionExpiration': {u'NoncurrentDays': 1},
+                 'Filter': {u'Prefix': ''},
+                 'Expiration': {u'Days': 1},
+                 'AbortIncompleteMultipartUpload': {u'DaysAfterInitiation': 1},
+                 'ID': 'cloudInquisitor'}
+            ]
+        }
 
         objects = list(bucket.objects.limit(count=1))
         versions = list(bucket.object_versions.limit(count=1))
         if not objects and not versions:
             bucket.delete()
             logger.info('Deleted s3 bucket {} in {}'.format(resource.resource_id, resource.account))
-
-        cinqPolicyApplied = False
-        for rule in bucket.LifecycleConfiguration().rules:
-            if rule['ID'] == 'cloudInquisitor':
-                cinqPolicyApplied = True
-        if not cinqPolicyApplied:
-            bucket.LifecycleConfiguration().put(lifecyclePolicy)
-            logger.info('Added policy to delete bucket contents in s3 bucket {} in {}'.format(resource.resource_id, resource.account))
+        else:
+            for rule in bucket.LifecycleConfiguration().rules:
+                if rule['ID'] == 'cloudInquisitor':
+                    break
+            else:
+                bucket.LifecycleConfiguration().put(lifecycle_policy)
```

Comment:
@HTTP500 I made one other mistake on this line -- it should be 
                `bucket.LifecycleConfiguration().put(LifecycleConfiguration=lifecycle_policy)`

Good catch on all of this btw, these changes needed to be made prior to turning on the S3 auditor.

Reply:
Fixed. Thanks :D